frontend/src/data/variables/CdcCovidProvider.test.ts

```diff
@@ -0,0 +1,352 @@
+import { WHITE_NH, TOTAL } from "../utils/Constants";
+import { MetricId } from "../config/MetricConfig";
+import { excludeTotal } from "../query/BreakdownFilter";
+
+function covidAndAcsRows(
+  fips: FipsSpec,
+  race: string,
+  cases: number | null,
+  deaths: number | null,
+  hosp: number | null,
+  population: number
+) {
+  return [
+    {
+      state_fips: fips.code,
+      state_name: fips.name,
+      cases: cases,
+      death_y: deaths,
+      hosp_y: hosp,
+      race_and_ethnicity: race,
+      population: population,
+    },
+    {
+      state_fips: fips.code,
+      state_name: fips.name,
+      race_and_ethnicity: race,
+      population: population,
+    },
+  ];
+}
+
+function covidAndCountyAcsRows(
+  fips: FipsSpec,
+  race: string,
+  cases: number | null,
+  deaths: number | null,
+  hosp: number | null,
+  population: number
+) {
+  return [
+    {
+      county_fips: fips.code,
+      county_name: fips.name,
+      cases: cases,
+      death_y: deaths,
+      hosp_y: hosp,
+      race_and_ethnicity: race,
+      population: population,
+    },
+    {
+      county_fips: fips.code,
+      county_name: fips.name,
+      race_and_ethnicity: race,
+      population: population,
+    },
+  ];
+}
+
+const METRIC_IDS: MetricId[] = [
+  "covid_cases",
+  "covid_cases_per_100k",
+  "covid_cases_pct_of_geo",
+  "covid_cases_reporting_population",
+  "covid_cases_reporting_population_pct",
+];
+
+export async function evaluateWithAndWithoutTotal(
+  covidDatasetId: string,
+  rawCovidData: any[],
+  acsDatasetId: string,
+  rawAcsData: any[],
+  baseBreakdown: Breakdowns,
+  breakdownVar: BreakdownVar,
+  nonTotalRows: any[],
+  totalRows: any[]
+) {
+  const acsProvider = new AcsPopulationProvider();
+  const cdcCovidProvider = new CdcCovidProvider(acsProvider);
+
+  dataFetcher.setFakeDatasetLoaded(covidDatasetId, rawCovidData);
+  dataFetcher.setFakeDatasetLoaded(acsDatasetId, rawAcsData);
+
+  // Evaluate the response with requesting total field
+  const responseWithTotal = await cdcCovidProvider.getData(
+    new MetricQuery(METRIC_IDS, baseBreakdown.addBreakdown(breakdownVar))
+  );
+  expect(responseWithTotal).toEqual(
+    new MetricQueryResponse(totalRows, [covidDatasetId, acsDatasetId])
+  );
+
+  // Evaluate the response without requesting total field
+  const responseWithoutTotal = await cdcCovidProvider.getData(
+    new MetricQuery(
+      METRIC_IDS,
+      baseBreakdown.addBreakdown(breakdownVar, excludeTotal())
+    )
+  );
+  expect(responseWithoutTotal).toEqual(
+    new MetricQueryResponse(nonTotalRows, [covidDatasetId, acsDatasetId])
+  );
+}
+
+autoInitGlobals();
+const dataFetcher = getDataFetcher() as FakeDataFetcher;
+
+describe("cdcCovidProvider", () => {
+  beforeEach(() => {
+    resetCacheDebug();
+    dataFetcher.resetState();
+    dataFetcher.setFakeMetadataLoaded(FakeDatasetMetadataMap);
+  });
+
+  test("County and Race Breakdown", async () => {
```

Comment:
You might consider also adding age & sex breakdown tests

Reply:
added more tests